src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -532,6 +532,14 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 		});
 
 		// Window close
+		this._win.once('close', () => {
```

Comment:
A window can veto the `close` so you maybe deleting storage that is in use: https://github.com/microsoft/vscode/blob/1f148906f1ece5c83a60ed8e211c9f387188dca3/src/vs/platform/lifecycle/electron-main/lifecycleMainService.ts#L395